nomad/node_endpoint.go

```diff
@@ -468,6 +476,18 @@ func (n *Node) UpdateDrain(args *structs.NodeUpdateDrainRequest,
 		args.DrainStrategy.ForceDeadline = time.Now().Add(args.DrainStrategy.Deadline)
 	}
 
+	// Construct the node event
+	args.NodeEvent = structs.NewNodeEvent().SetSubsystem(structs.NodeEventSubsystemDrain)
+	if node.DrainStrategy == nil && args.DrainStrategy == nil {
+		return nil // Nothing to do
```

Comment:
Shouldn't this go to line 492 where you apply the node update raft transaction? If there is no node drain event, seems like it should still apply the node update without the event.

Reply:
So I made this skip the raft transaction since it is a no-op. It is essentially saying, please update the node to have no drain strategy even when the node already doesn't have one.